src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -373,7 +373,7 @@ SegmentCleaner::gc_reclaim_space_ret SegmentCleaner::gc_reclaim_space()
     ceph_assert(!scan_cursor->is_complete());
   }
 
-  return scanner->scan_extents(
+  return scanner->scan_extents<true>(
```

Comment:
> I'm not sure if it is meaningful to sort extents **within the same reclaim transaction**. Could you add more explainations?

This is following the sprite lfs paper.

Reply:
If you are referring to the "age sort" concept from the paper. IIUC it is referring to the process to relocate and group the extent of the similar age **into the same segment**, which is not related to sorting the extents by age **within the same transaction**.

Reply:
Um, I don't think so. The paper explicitly stated that ` the live blocks were sorted by age before writing them out again` which is later refered to as "age sort". Actually, I think we can consider it this way: what is the "similarity" degree that we should consider as the threshold to put extents in the same segment? This is not a easy question, the answer is related to the actual clients' I/O pattern. The easiest way that we can take is to sort the extents out and, as time goes by, extents' age difference within the same segment will converge to relatively "similar"

Reply:
Modifying scan_extents like this will only sort the output of a single bytes_to_read scan, right?  Those extents will then be written out in a single transaction regardless of order.  It's not clear to me how that could help.  The paper's goal with sorting by age is to ensure that extents written together are approximately the same age -- this doesn't seem to accomplish that.  I'd like to drop this portion for now and revisit it later.

Reply:
> Those extents will then be written out in a single transaction regardless of order

I don't quite follow this, extents will be written in the order of the result of the scan, right?

Reply:
Yes, but order of extents within a transaction seems irrelevant.

Reply:
Ah, yes.